### ISSUE TITLE
KeyFilter: regex does not filter non-digit if typed right after accent

### ISSUE THREAD

**Issue by mateusscheper** [primefaces/primefaces#9818]
### Describe the bug

<p:keyFilter> regex does not filter non-digit if typed right after accent. 
Example:
```
<p:inputText id="myId"
    styleClass="myClass"
    value="#{bean.value}"
    disabled="#{bean.disable}">
        <p:ajax process="@this" global="false"/>
        <p:keyFilter regEx="/[a-zA-Z0-9]/" preventPaste="false"/>
</p:inputText>
```
If I type "~" and "]" repeatedly, they won't be filtered. Same goes for other non-digit like "@", "´", etc.

### Reproducer

```
<p:inputText id="myId"
    styleClass="myClass"
    value="#{bean.value}"
    disabled="#{bean.disable}">
        <p:ajax process="@this" global="false"/>
        <p:keyFilter regEx="/[a-zA-Z0-9]/" preventPaste="false"/>
</p:inputText>
```

### Expected behavior

Since regex filters correctly if I type characters like "@" and "[", it should also filter them if I type after an accent, like "~".

### PrimeFaces edition

Community

### PrimeFaces version

11.0.6

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

11

### Browser(s)

Tested on Chrome 110.0.5481.100

**Comment by melloware** [comment]
I will investigate this.

**Comment by melloware** [comment]
Maybe I am doing it wrong but I can't reproduce.  Attached is a reproducer just run `mvn clean jetty:run` and navigate to localhost:8080/primefaces-test/

[pf-9818.zip](https://github.com/primefaces/primefaces/files/10794276/pf-9818.zip)

**Comment by mateusscheper** [comment]
I managed to reproduce the issue. Tested on PrimeFaces 12 and 11.0.6. [Gif here](https://giphy.com/gifs/W3HAbw4EEOb2JTNr0J)

**Comment by melloware** [comment]
I am thinking it might be I use obviously English US keyboard. @newk5 any thoughts here with PT_BR keyboard?

**Comment by melloware** [comment]
Yep here: https://www.primefaces.org/showcase/ui/input/keyFilter.xhtml?jfwid=33923  The REGEX one only should accept a b or c

**Comment by newk5** [comment]
Yea seems to be working correctly for me. I can't type any accents. In my keyboard layout typing an accent involves:
( SHIFT+Accent key ) + letter. I dont know if its the same for PT_BR layout

**Comment by melloware** [comment]
Its weird all the code is doing isa Javascript regex test 

```js
/[a-zA-Z0-9]/.test('~');
```

If I enter that in my F12 console I get the correct `false` result.

![image](https://user-images.githubusercontent.com/4399574/220458550-62022c39-89c1-4ae7-bb2d-a35d4bb3fae3.png)

**Comment by mateusscheper** [comment]
Just an update: I tested yesterday on Windows and the issue didn't happen. Doing a little research today, I found out that this issue is most likely due to dead keys behavior on Linux Mint (where I'm seeing the issue): https://forums.linuxmint.com/viewtopic.php?t=380772

I changed my keyboard layout to pt-br with no dead keys as a test and the validation worked as expected. 
Since it involves a particular feature of Linux Mint (I'm using Linux Mint 21 Cinnamon), I think we could let it go.
